packages/lumx-react/src/components/alert-dialog/AlertDialog.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { forwardRef } from 'react';
+
+import classNames from 'classnames';
+
+import {
+    FlexBox,
+    DialogProps,
+    Dialog,
+    Button,
+    Emphasis,
+    ColorPalette,
+    Icon,
+    Size,
+    Alignment,
+    Kind,
+    Orientation,
+} from '@lumx/react';
+
+import { mdiAlert, mdiAlertCircle, mdiCheckCircle, mdiInformation } from '@lumx/icons/';
+import { uid } from 'uid';
+import { Comp, getRootClassName, handleBasicClasses } from '@lumx/react/utils';
+
+export interface AlertDialogProps extends Omit<DialogProps, 'header' | 'footer'> {
+    /** Message variant. */
+    kind?: Kind;
+    /** Dialog title. Can be either a string, or sanitized html. */
+    title?: string;
+    className?: string;
+    /** The props to the confirm button */
+    confirmProps: {
+        onClick(): void;
+        label: string;
+    };
+    /**
+     * The props to the cancel button.
+     * Will not render a cancel button if undefined.
+     */
+    cancelProps?: {
+        onClick(): void;
+        label: string;
+    };
+}
+
+/**
+ * Associative map from message kind to color and icon.
+ */
+const CONFIG = {
+    [Kind.error]: { color: ColorPalette.red, icon: mdiAlert },
+    [Kind.info]: { color: ColorPalette.blue, icon: mdiInformation },
+    [Kind.success]: { color: ColorPalette.green, icon: mdiCheckCircle },
+    [Kind.warning]: { color: ColorPalette.yellow, icon: mdiAlertCircle },
+};
+
+/**
+ * Component display name.
+ */
+const COMPONENT_NAME = 'AlertDialog';
+
+/**
+ * Component default class name and class prefix.
+ */
+const CLASSNAME = getRootClassName(COMPONENT_NAME);
+
+/**
+ * Component default props.
+ */
+const DEFAULT_PROPS: Partial<DialogProps> = {
+    size: Size.tiny,
+    kind: Kind.info,
+};
+
+/**
+ * AlertDialog component.
+ *
+ * An alert dialog is a modal dialog that interrupts the user's workflow to
+ * communicate an important message and acquire a response.
+ *
+ * It should not have a complexe content.
+ * Children of this component should only be strings, paragraphs or links.
+ */
+export const AlertDialog: Comp<AlertDialogProps, HTMLDivElement> = forwardRef(
+    (
+        { id, title, className, cancelProps, confirmProps, kind, size, dialogProps = {}, children, ...forwardedProps },
```

Comment:
I keep it to you appreciation but we tend to avoid de-structuring long props list in the argument list and move it in the component body

Reply:
Also it's not necessary to initialize `dialogProps` because it's valid to spread an `undefined` value.

Reply:
Done and done ;)